src/cli/programutilities.cpp

```diff
@@ -18,4 +23,121 @@ QString loadFlatBinaryFile(Program &program, const QString &filepath,
   return QString();
 }
 
+using namespace ELFIO;
+class ELFIODwarfLoader : public ::dwarf::loader {
+public:
+  ELFIODwarfLoader(elfio &reader) : reader(reader) {}
+
+  const void *load(::dwarf::section_type section, size_t *size_out) override {
+    auto sec = reader.sections[::dwarf::elf::section_type_to_name(section)];
+    if (sec == nullptr)
+      return nullptr;
+    *size_out = sec->get_size();
+    return sec->get_data();
+  }
+
+private:
+  elfio &reader;
+};
+
+std::shared_ptr<ELFIODwarfLoader> createDwarfLoader(elfio &reader) {
```

Comment:
Nit: (i know this is my own code) - I think it's best practice to add `static` to `.cpp`-local function definitions to ensure that they are indeed private to the file - see https://llvm.org/docs/CodingStandards.html#anonymous-namespaces.